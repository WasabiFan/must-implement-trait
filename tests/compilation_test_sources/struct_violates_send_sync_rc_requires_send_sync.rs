use std::rc::Rc;
use must_implement_trait::must_implement_trait;

fn main() {
    let _instance = MySendSyncStruct { value: 7 };
}

#[must_implement_trait(Send, Sync)]
struct MySendSyncStruct {
    value: Rc<i32>
}