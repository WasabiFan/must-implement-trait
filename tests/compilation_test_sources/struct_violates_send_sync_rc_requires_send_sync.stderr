error[E0277]: `std::rc::Rc<i32>` cannot be shared between threads safely
 --> $DIR/struct_violates_send_sync_rc_requires_send_sync.rs:8:1
  |
8 | #[must_implement_trait(Send, Sync)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `std::rc::Rc<i32>` cannot be shared between threads safely
  |
  = help: within `MySendSyncStruct`, the trait `std::marker::Sync` is not implemented for `std::rc::Rc<i32>`
  = note: required because it appears within the type `MySendSyncStruct`
  = help: see issue #48214
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `std::rc::Rc<i32>` cannot be sent between threads safely
 --> $DIR/struct_violates_send_sync_rc_requires_send_sync.rs:8:1
  |
8 | #[must_implement_trait(Send, Sync)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `std::rc::Rc<i32>` cannot be sent between threads safely
  |
  = help: within `MySendSyncStruct`, the trait `std::marker::Send` is not implemented for `std::rc::Rc<i32>`
  = note: required because it appears within the type `MySendSyncStruct`
  = help: see issue #48214
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
